vault/dynamic_system_view.go

```diff
@@ -136,6 +136,20 @@ func (e extendedSystemViewImpl) SudoPrivilege(ctx context.Context, path string,
 	return authResults.RootPrivs
 }
 
+func (e extendedSystemViewImpl) APILockShouldBlockRequest() (bool, error) {
+	mountEntry := e.mountEntry
+	if mountEntry == nil {
+		return false, fmt.Errorf("no mount entry")
+	}
+	ns := mountEntry.Namespace()
+
+	if err := enterpriseBlockRequestIfError(e.core, ns.Path, mountEntry.Path); err != nil {
```

Comment:
My only concern with this PR was (I think) that this code was previously only being called with the stateLock held, i.e. as part of regular request handling.  Now we're calling it without that lock.  From what I can tell that won't be a problem, just calling it out so you're aware, in case it helps with potential future issues.